modules/update-process-workflow.adoc

```diff
@@ -0,0 +1,48 @@
+// Module included in the following assemblies:
+//
+// * updating/understanding_updates/how-updates-work.adoc
+
+:_content-type: CONCEPT
+[id="update-process-workflow_{context}"]
+= Update process workflow
+
+The following represents a detailed workflow of the {product-title} update process.
+
+. The intent to update the cluster is expressed through the web console or the CLI.
+The target version is stored in the `spec.desiredUpdate.version` field of the ClusterVersion resource.
+
+. The Cluster Version Operator (CVO) discovers that the `desiredUpdate` in the ClusterVersion resource differs from the current cluster version.
+Using the graph data from the OpenShift Update Service, the CVO resolves the desired cluster version to a pullspec to the release image.
+
+. The CVO validates the integrity and authenticity of the release image.
+Red Hat publishes cryptographically-signed statements about published release images at predefined locations, using image SHA digests as unique and immutable release image identifiers.
+The CVO utilizes a list of built-in public keys to validate the presence and signatures of the statement matching the checked release image.
+
+. The CVO creates a job named `version-$version-$hash` in the `openshift-cluster-version` namespace.
+This job uses containers that are executing the release image, so the cluster first downloads the image through the container runtime.
+The job then extracts the manifests and metadata from the release image to a shared volume that is accessible to the CVO.
+
+. The CVO validates the extracted manifests and metadata.
+
+. The CVO ensures that no problematic condition is observed in the cluster, such as `ClusterOperator` reporting a `Failing=True` condition.
+Certain conditions can prevent updates from proceeding.
+These conditions are either determined by the CVO itself, or reported by individual Cluster Operators that observe some details about the cluster that they consider problematic for the update.
+
+. The CVO records the accepted release in `status.desired` and creates a `status.history` entry about the new update.
+
+. The CVO begins applying the manifests from the release image.
+Cluster Operators are updated in separate stages called Runlevels, and the CVO ensures that all Operators in a Runlevel finish updating before it proceeds to the next level.
+
+. Manifests for the CVO itself are applied towards the beginning of the order, so the CVO is updated early in the process.
+When the CVO deployment is applied, the current CVO Pod terminates, and a CVO Pod using the new version starts.
+The new CVO proceeds to apply the remaining manifests.
```

Comment:
nit: it might just be me, but "apply" feels like a one-off transition.  In actuality, the CVO is repeatedly looping through its manifest set attempting to reconcile each manifest to an acceptably-happy state.  Perhaps some of the "apply" in this pull request could be migrated to "reconcile", with the occasional "apply" reserved for things that are specific to the transitional sub-case (like "When the CVO deployment is applied, the current CVO Pod terminates...")?  Feel free to ignore this comment if I'm not making sense.